packages/core/lib/commands/test/run.js

```diff
@@ -68,19 +68,50 @@ module.exports = async function (options) {
     inputFile: file
   });
 
-  if (config.networks[config.network]) {
+  /**
+   * If test network exists and provider, host or url doesn't exist, then internal ganache is started with the user specified
+   * configurations in the config.
+   * If development or test network with provider, host or url exists, then it connects to that network specified in the config.
+   * If no network is specified in the config, then internal ganache is started with default configuration.
+   */
+  const configuredNetwork = config.networks[config.network];
+  const testNetworkDefined = configuredNetwork && config.network == "test";
+  const noProviderHostOrUrlConfigured =
+    configuredNetwork &&
+    !configuredNetwork.provider &&
+    !configuredNetwork.host &&
+    !configuredNetwork.url;
+  const port = await require("get-port")();
```

Comment:
We probably shouldn't be getting a port unless we're actually in the managed case; I don't know how much work it actually takes to get a port, but it strikes me as probably something to be avoided in the cases where we don't need it.